File: Mixter.Domain/Core/Subscriptions/Events/UserUnfollowed.cs
namespace Mixter.Domain.Core.Subscriptions.Events
{
    [Event]
    public struct UserUnfollowed : IDomainEvent
    {
        public SubscriptionId SubscriptionId { get; private set; }

        public UserUnfollowed(SubscriptionId subscriptionId) : this()
        {
            SubscriptionId = subscriptionId;
        }

        public object GetAggregateId()
        {
            return SubscriptionId;
        }
    }
}

File: domain/src/main/java/mixter/domain/core/subscription/events/UserUnfollowed.java
package mixter.domain.core.subscription.events;

import mixter.domain.AggregateId;
import mixter.domain.Event;
import mixter.domain.core.subscription.SubscriptionId;

public class UserUnfollowed implements Event {
    private SubscriptionId subscriptionId;

    public UserUnfollowed(SubscriptionId subscriptionId) {
        this.subscriptionId = subscriptionId;
    }

    @Override
    public AggregateId getId() {
        return subscriptionId;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        UserUnfollowed that = (UserUnfollowed) o;

        return !(subscriptionId != null ? !subscriptionId.equals(that.subscriptionId) : that.subscriptionId != null);

    }

    @Override
    public int hashCode() {
        return subscriptionId != null ? subscriptionId.hashCode() : 0;
    }

    public SubscriptionId getSubscriptionId() {
        return subscriptionId;
    }
}